hypershift-operator/controllers/hostedcluster/network_policies.go

```diff
@@ -332,12 +332,106 @@ func reconcilePrivateRouterNetworkPolicy(policy *networkingv1.NetworkPolicy, hcl
 			},
 		},
 	}
+
+	clusterNetworks := make([]string, 0)
+	// In vanilla kube management cluster this would be nil.
+	if managementClusterNetwork != nil {
+		for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+			clusterNetworks = append(clusterNetworks, network.CIDR)
+		}
+	}
+
+	// Allow to any destination not on the management cluster service network
+	policy.Spec.Egress = []networkingv1.NetworkPolicyEgressRule{
```

Comment:
Should we cover IPv6 also in this PR? Could be done in a follow up one? same as we do in https://github.com/openshift/hypershift/blob/7aad3c82e96817443053ac3b183b6c1864b657ac/hypershift-operator/controllers/hostedcluster/network_policies.go#L241

Reply:
I would rather we merged this as it is now, because it is needed for the ROSA security hardening epic.
We can have a follow-up for disconnected/ipv6 if it is needed